File: Cpp/Longest-Increasing-Continuous-subsequence.cpp
#include <vector>

class Solution {
public:
    /**
     * @param A an array of Integer
     * @return  an integer
     */
    int longestIncreasingContinuousSubsequence(vector<int>& A) {
        // Write your code here
        if (A.empty()) return 0;
        int len = A.size(), m = 1, res = 1;
        for (int i = 1; i < len; i ++) {
            if (A[i] > A[i-1]) {
                m += 1;
                res = max(res, m);
            } else {
                m = 1;
            }
        }
        m = 1;
        for (int i = len - 2; i >= 0; i --) {
            if (A[i] > A[i+1]) {
                m += 1;
                res = max(res, m);
            } else {
                m = 1;
            }
        }
        return res;
    }
};

File: Java/Longest-Increasing-Continuous-subsequence.java
public class Solution {
    /**
     * @param A an array of Integer
     * @return  an integer
     */
    public int longestIncreasingContinuousSubsequence(int[] A) {
        // Write your code here
        if (A == null || A.length <= 0) return 0;
        int len = A.length;
        int maxlen = 1, increlen = 1, decrelen = 1;
        for (int i = 1; i < len; i ++) {
            if (A[i] >= A[i - 1]) {
                increlen ++;
            } else {
                increlen = 1;
            }
            if (A[i] <= A[i - 1]) {
                decrelen ++;
            } else {
                decrelen = 1;
            }
            maxlen = Math.max(maxlen, Math.max(decrelen, increlen));
        }
        return maxlen;
    }
}
